docs/get_started/chat_agent.md

```diff
@@ -0,0 +1,206 @@
+
+## Using the Methods of the `ChatAgent` Class
+
+The `ChatAgent` class provides a suite of methods that enable the agent to manage conversations, generate responses, and perform various operations related to chat interactions. Here's an exploration of some key methods:
+
+1. **Resetting the Agent**:
+    The `reset` method allows you to reset the `ChatAgent` to its initial state and also retrieve the stored messages.
+    ```python
+    stored_messages = agent.reset()
+    print(stored_messages)
+    >>> [<BaseMessage object>, ...]
+    ```
+
+2. **Setting and Getting the System Message**:
+    Use the setter and getter methods to update and retrieve the system message of the agent.
+    ```python
+    # Setting the system message
+    agent.system_message = new_system_message
+
+    # Getting the system message
+    current_system_message = agent.system_message
+    print(current_system_message)
+    >>> <BaseMessage content="Your system message">
+    ```
+
+3. **Checking Function Calling Capability**:
+    The method `is_function_calling_enabled` informs if the OpenAI function calling is enabled for the agent.
+    ```python
+    is_enabled = agent.is_function_calling_enabled()
+    print(is_enabled)
+    >>> True
+    ```
+
+4. **Setting Output Language**:
+    The `set_output_language` method allows you to specify the language for the generated output and retrieve the updated system message.
+    ```python
+    updated_message = agent.set_output_language("en")
+    print(updated_message)
+    >>> <BaseMessage content="Your updated system message">
+    ```
+
+5. **Retrieving Chat Session Information**:
+    Use the `get_info` method to obtain information about the chat session, including ID, usage, termination reasons, and more.
+    ```python
+    info = agent.get_info(id="12345", usage={}, termination_reasons=[], num_tokens=100, called_funcs=[])
+    print(info)
+    >>> {"id": "12345", "usage": {}, ...}
+    ```
+
+6. **Initializing Stored Messages**:
+    The `init_messages` method initializes the stored messages list with the initial system message.
+    ```python
+    agent.init_messages()
+    print(agent.stored_messages)
+    >>> [<BaseMessage object>, ...]
+    ```
+
+7. **Updating Stored Messages**:
+    The `update_messages` method allows you to add a new message to the stored messages and retrieve the updated list.
+    ```python
+    updated_messages = agent.update_messages(role="user", message=new_message)
+    print(updated_messages)
+    >>> [<BaseMessage object>, ...]
+    ```
+
+8. **Submitting an External Message as an Assistant Response**:
+    The `submit_message` method allows you to provide an external message as if it were a response from the chat LLM.
+    ```python
+    external_message = BaseMessage(...)
+    agent.submit_message(external_message)
+    ```
+
+9. **Performing a Chat Session Step**:
```

Comment:
This is the most important member function of ChatAgent for most of the users. Please put it on the top of the file.

Reply:
> This is the most important member function of ChatAgent for most of the users. Please put it on the top of the file.

done